app/app.go

```diff
@@ -587,9 +587,6 @@ func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.Res
 	if err := tmjson.Unmarshal(req.AppStateBytes, &genesisState); err != nil {
 		panic(err)
 	}
-	if req.ConsensusParams != nil && req.ConsensusParams.Version != nil {
-		app.SetProtocolVersion(req.ConsensusParams.Version.AppVersion)
-	}
```

Comment:
why is this no longer necessary?

Reply:
It's done in the SDK instead

Reply:
See https://github.com/celestiaorg/cosmos-sdk/blob/18e768c968e05660e4ad0a7f4cc3ea8a33d5c993/baseapp/abci.go#L64

Reply:
Assuming it can be done in either location (app or SDK), why do it in the SDK where it has the risk of getting overwritten when we pull upstream changes?

Reply:
Because this behaviour is logically the responsibility of baseapp not app

Reply:
Then can it be upstreamed to cosmos/cosmos-sdk so it doesn't accidentally get reverted in celestiaorg/cosmos-sdk when we pull upstream changes?